llvm/lib/Transforms/Scalar/LoopInterchange.cpp

```diff
@@ -276,11 +320,12 @@ static bool isLegalToInterChangeLoops(CharMatrix &DepMatrix,
       continue;
 
     // Check if the direction vector is lexicographically positive (or zero)
-    // for both before/after exchanged.
-    if (isLexicographicallyPositive(Cur, OuterLoopId, Cur.size()) == false)
+    // for both before/after exchanged. Ignore the last element because it
+    // doesn't affect the legality.
```

Comment:
[not a change request] This is probably true for anything inside `InnerLoopId` as well, but historically we checked the inner vector as well, like we do check dependencies before the interchange (which should tautologically be non-negative depdencence vectors)